pkg/pillar/types/zfs.go

```diff
@@ -85,6 +153,8 @@ type ZFSPoolStatus struct {
 	Disks            []*StorageDiskState
 	CollectorErrors  string
 	Children         []*StorageChildren
+	Metrics          *ZDeviceMetrics // metrics and error counters for zpool
+	PoolStatusMsg    PoolStatus      // pool status value from ZFS
```

Comment:
Are we going to report the metrics to the controller using the metrics API i.e., periodic reports? I think we report the ZFSPoolStatus content using the event-triggered info API. Depending on the answer it might make sense to have a separate ZFSPoolMetrics struct which will contain the things which will be reported periodically.

Reply:
Yes, it is in PR: #2648 
ZFSPoolStatus serves as a container in which we collect Zpool data. And in the information API there is no subscription to events its updates (Create, modify, delete). There, when calling, we get all available ZFSPoolStatus, collect an informational message from the data, and that's it. Therefore, I think that there is no need to move Metrics from ZFSPoolStatus to a separate ZFSPoolMetrics. But if this is still critical for us, let me know about it, and I change it.

Another question is how often we want to collect these metrics. After @pram answers this question, I can add a new go-routine at the right interval to update this information.

Reply:
The philosophy is that Status changes due to some event (which could be either some configuration change, or some internally generated event such as an error), and as such are reported through the system (publish towards zedagent and then zedagent sending an Info message to the controller when such an event/change takes place.

Metrics are always periodic; collected and sent periodically.

The fact that you say how often we want to collect this tells me is should be a separate metrics struct.

Reply:
Сorrected